app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= current_user # guest user (not logged in)
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :read, :all
+      can :create, [Post, Comment]
+      can :update, [Post, Comment], author_id: user.id
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
+    end
```

Comment:
- [ ] Great job adding the ability class, unfortunately, I couldn't create a post using a user or admin role. Kindly fix the issue before submitting the PR for review. To test this project a post, comment and a like should be create.

![authorization](https://github.com/ritikarawat220/Blog-App/assets/53124548/6656e0cb-312d-44e3-ba0b-d0f31121bc26)

Reply:
Sorry I approved the PR instead of requested review, the dashboard will show changes required.

Reply:
It seems the issue of creating the post on behalf of logged in user still persists. Take a look at the below screenshot of the console. ⬇️
![image](https://github.com/ritikarawat220/Blog-App/assets/33268927/fc23a7eb-d76d-499b-8e6b-aa0bf380f451)

Here is the screen recording of the application. ⬇️


https://github.com/ritikarawat220/Blog-App/assets/33268927/00adc4a6-11a3-47cc-92d9-75da52c06046